tools/gengen/util/gengen.go

```diff
@@ -121,31 +119,32 @@ func GenGen(ctx context.Context, cfg *GenesisCfg, cst *hamt.CborIpldStore, bs bl
 	}
 
 	st := state.NewTree(cst)
-	storageMap := vm.NewStorageMap(bs)
+	store := vm.NewStorage(bs)
+	vm := vm.NewVM(st, &store).(consensus.GenesisVM)
 
 	if err := actor.InitBuiltinActorCodeObjs(cst); err != nil {
 		return nil, err
 	}
 
-	if err := consensus.SetupDefaultActors(ctx, st, storageMap, cfg.ProofsMode, cfg.Network); err != nil {
+	if err := consensus.SetupDefaultActors(ctx, vm, &store, st, cfg.ProofsMode, cfg.Network); err != nil {
 		return nil, err
 	}
 
-	if err := setupPrealloc(ctx, st, storageMap, keys, cfg.PreAlloc); err != nil {
+	if err := setupPrealloc(ctx, vm, st, keys, cfg.PreAlloc); err != nil {
 		return nil, err
 	}
 
-	miners, err := setupMiners(st, storageMap, keys, cfg.Miners, pnrg)
+	miners, err := setupMiners(vm, st, keys, cfg.Miners, pnrg)
 	if err != nil {
 		return nil, err
 	}
 
-	stateRoot, err := st.Flush(ctx)
+	err = store.Flush()
 	if err != nil {
 		return nil, err
```

Comment:
This reordering is a nice catch. Weird this has always worked.  I guess small genesis size and hamt width has saved us so far?

Reply:
There are a few things in this file that fall under the "not sure how this is working" category.